### ISSUE TITLE
Enable pluggable networking

### ISSUE THREAD

**Issue by cblgh** [cabal-club/cabal-client#13]
Just a thought: eventually it'd be great if we could move out the networking elsewhere, maybe a pluggable set of modules, so that regardless of whether you're doing node/electron/web/mobile, you can use the discovery methods that work for that platform.

_Originally posted by @noffle in https://github.com/_render_node/MDE3OlB1bGxSZXF1ZXN0UmV2aWV3Mjc2Mjg4OTYz/pull_request_reviews/more_threads_

**Comment by okdistribute** [comment]
+1
maybe this could be related to the work by @telamohn

**Comment by telamon** [comment]
Yes this sounds like the same thing that i want to do, we're not there yet though. 
@cblgh the link you posted dosen't seem to work :/

**Comment by cblgh** [comment]
@telamon i don't know why it doesn't work, but it's from  https://github.com/cabal-club/cabal-client/pull/11

the comment contains everything (except i guess the context of the particular code snippet) that noffle originally posted
